client/components/ChallengesList.js

```diff
@@ -26,12 +33,45 @@ const component = ({
   useEffect(() => {
     if (difficulty) {
       filter = difficulty;
+      fetchAllUserchallenges();
```

Comment:
This is called regardless of whether `difficulty` exists, so move outside of if statement.
```suggestion
```